### PR TITLE
Bluetooth: sample: Add ISO support to hci_uart

### PR DESCRIPTION
**Pull request by Thalley**
Add ISO support to the hci_uart sample.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/hci_uart/src/main.c [hunk 1]
```diff
@@ -70,22 +71,39 @@ static int h4_read(const struct device *uart, uint8_t *buf, size_t len)
 
 static bool valid_type(uint8_t type)
 {
-	return (type == H4_CMD) | (type == H4_ACL);
+	return (type == H4_CMD) | (type == H4_ACL) | (type == H4_ISO);
 }
 
-/* Function assumes that type is validated and only CMD or ACL will be used. */
+/* Function expects that type is validated and only CMD, ISO or ACL will be used. */
```
Comment (koffes) [inline samples/bluetooth/hci_uart/src/main.c:77]: Nitpick.com: Would be nice if CMD, ACL and ISO are referenced in the same order on line 74, 77 and the switch on 80 and 97

**Review by koffes — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/hci_uart/src/main.c:77]: Not sure if you want any specific order, but it should be ordered as CMD, ISO, ACL everywhere now :)

**Review by Casper-Bonde-Bose — APPROVED** [review]
LGTM
Maybe change the build configuration for this application to enable the ISO feature by default? It takes a while to understand that failing to setup an ISO stream is due to wrong application configuration :-)

**Comment by Thalley** [comment]
> LGTM
> Maybe change the build configuration for this application to enable the ISO feature by default? It takes a while to understand that failing to setup an ISO stream is due to wrong application configuration :-)

That's not a bad idea. Do you think we should add it as a unicast server (only peripheral is enabled) or broadcast source?

**Comment by Casper-Bonde-Bose** [comment]
> > LGTM
> > Maybe change the build configuration for this application to enable the ISO feature by default? It takes a while to understand that failing to setup an ISO stream is due to wrong application configuration :-)
> 
> That's not a bad idea. Do you think we should add it as a unicast server (only peripheral is enabled) or broadcast source?

As it is a HCI build the host stack will not be running on the device/core that is running the uart_hci app, hence I believe there should be enough resource to enable everything.

**Pull request merged**